src/inventory/add/AddInventoryItem.tsx

```diff
@@ -0,0 +1,213 @@
+import { Button } from '@hospitalrun/components'
+import React, { useState, useEffect } from 'react'
+import { useHistory } from 'react-router-dom'
+
+import useAddBreadcrumbs from '../../page-header/breadcrumbs/useAddBreadcrumbs'
+import { useUpdateTitle } from '../../page-header/title/TitleContext'
+import SelectWithLabelFormGroup, {
+  Option,
+} from '../../shared/components/input/SelectWithLabelFormGroup'
+import TextFieldWithLabelFormGroup from '../../shared/components/input/TextFieldWithLabelFormGroup'
+import TextInputWithLabelFormGroup from '../../shared/components/input/TextInputWithLabelFormGroup'
+import useTranslator from '../../shared/hooks/useTranslator'
+import InventoryItem from '../../shared/model/InventoryItem'
+import useAddInventoryItem from '../hooks/useAddInventoryItem'
+import { InventoryItemError } from './validate-inventory-item'
+
+const AddInventoryItem = () => {
+  const [mutate] = useAddInventoryItem()
+  const { t } = useTranslator()
+  const history = useHistory()
+  const updateTitle = useUpdateTitle()
+  useEffect(() => {
+    updateTitle(t('inventory.items.new'))
+  })
+  const [error, setError] = useState<InventoryItemError | undefined>(undefined)
+
+  const [addInventoryItem, setAddInventoryItem] = useState(({
+    name: '',
+    rank: '',
+    type: '',
+    crossReference: '',
+    reorderPoint: ('' as unknown) as number,
```

Comment:
This casting feels strange to me. Why not start with a number?

Reply:
Since the value is set through a text input, it's a string. However, strings and numbers don't appear to overlap enough to simply cast from one to the other. Turns out you have to make it unknown first. I would try to change the input to a numerical one but I didn't see any templates on the inputs folder so I decided to leave it as is.